src/astarte/handler.rs

```diff
@@ -321,6 +320,44 @@ impl DevicePublisher {
 
         Ok(())
     }
+
+    // check if the interface is present in the node introspection
+    async fn find_in_introspection(
+        &self,
+        node_id: NodeId,
+        interface_name: &str,
+    ) -> Result<(), AstarteMessageHubError> {
+        let node = self.subscribers.read().await;
+        let iface_exists = node
+            .get(&node_id)
+            .ok_or_else(|| AstarteMessageHubError::NodeId(node_id.into()))?
+            .introspection
+            .contains(interface_name);
+
+        if !iface_exists {
+            return Err(AstarteMessageHubError::MissingInterface {
+                interface: interface_name.to_string(),
+                node_id: node_id.into(),
+            });
+        }
+
+        Ok(())
+    }
+
+    async fn retain_node_props(
```

Comment:
```suggestion
    async fn filter_node_props(
```

nit: I think it conveys the action more clearly, but it's not too important